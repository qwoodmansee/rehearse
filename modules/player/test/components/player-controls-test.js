import PlayerControls from '@player/src/components/player-controls';
import React from 'react';
import { ConsistentPlayerControlList, RandomPlayerControlList } from '@player/test/factories/player-control-factory';
import { fireEvent, render } from '@testing-library/react-native';

describe('PlayerControls', () => {
  it('renders', () => {
    const controls = ConsistentPlayerControlList();
    const { baseElement } = render(<PlayerControls controls={controls} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('correctly associates each button to their control function', () => {
    const controls = ConsistentPlayerControlList();
    const mockControlFunctionOne = jest.fn();
    const mockControlFunctionTwo = jest.fn();
    controls[1].controlFn = mockControlFunctionOne;
    controls[2].controlFn = mockControlFunctionTwo;
    const { getAllByTestId } = render(<PlayerControls controls={controls} />);

    fireEvent.press(getAllByTestId('internal-button')[1]);

    expect(mockControlFunctionOne).toHaveBeenCalled();
    expect(mockControlFunctionTwo).not.toHaveBeenCalled();
  });

  it('renders the correct number of buttons based on the number of controls', () => {
    const controls = RandomPlayerControlList(8);
    const { getAllByTestId } = render(<PlayerControls controls={controls} />);

    expect(getAllByTestId('internal-button').length).toEqual(8);
  });
});
