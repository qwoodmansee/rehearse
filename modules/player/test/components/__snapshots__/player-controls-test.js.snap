// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControls renders 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-evenly",
          "width": "90%",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f05545",
              "borderRadius": 20,
              "height": 80,
              "justifyContent": "center",
              "padding": 10,
              "position": "relative",
              "width": 80,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="internal-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../modules/player/assets/metronome-up.png",
            }
          }
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
